### DISCUSSION TITLE
Can you define a step to only perform once within a matrix?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#31799] [Actions]
I've set up an action where I use a matrix to publish certain assets towards a site, once for each supported platform.

The issue I encounter right now is, that the assets in question are created 4 separate times, once for each platform, which means I download the required dependencies 4 times and obviously build the assets 4 separate times.

I'm not that experienced with matrices, so Idk if you can define one or multiple steps to run once globally to then use in the different matrix instances. Is that possible?

If not, can I make it so that the assets, which btw are .jar files, would be temporarily saved within a job and then use in the next job with the matrix?

The currently used workflow file can be found here:
https://github.com/Andre601/AdvancedServerList/blob/master/.github/workflows/release.yml

**Comment by akostadinov** [comment]
The link is gone already. Old question. But for anybody searching , the only option I see so far is something like `if: matrix.item == 'whatevermatrixvalueyouhave'` so that it runs only for `whatevermatrixvalueyouhave`
